docs/src/testing/using-cheatcodes.md

```diff
@@ -174,32 +197,81 @@ Failures:
     package_name::call_and_invoke
 ```
 
-### Pranking the constructor
+### Mocking the constructor with `prank`
 
 Most of the cheatcodes like `prank`, `mock_call`, `warp`, `roll` do work in the constructor of the contracts.
 
 Let's say, that you have a contract that saves the caller address (deployer) in the constructor, and you want it to be pre-set to a certain value.
 
-To `prank` the constructor, you need to `start_prank` before it is invoked, with the right address. 
-To achieve this, you need to precalculate address of the contract using `precalculate_address` of `ContractClassTrait` on declared contract,
-and then use it in `start_prank` as an argument:
-
+To `prank` the constructor, you need to `start_prank` before it is invoked, with the right address. To achieve this, you need to precalculate the address of the contract by using the `precalculate_address` function of `ContractClassTrait` on the declared contract, and then use it in `start_prank` as an argument:
 
 ```rust
 use snforge_std::{ declare, ContractClassTrait, start_prank };
 
 #[test]
-fn prank_the_constructor() {
+fn mock_constructor_with_prank() {
+    // declaring contract
     let contract = declare('HelloStarknet');
-    let constructor_arguments = @ArrayTrait::new();
-    
     // Precalculate the address to obtain the contract address before the constructor call (deploy) itself
-    let contract_address = contract.precalculate_address(constructor_arguments); 
-    
+    let contract_address = contract.precalculate_address(@ArrayTrait::new());
     // Change the caller address to 123 before the call to contract.deploy
     start_prank(contract_address, 123.try_into().unwrap());
-    
-    // The constructor will have 123 set as the caller address 
+
+    // The constructor will have 123 set as the caller address
     contract.deploy(constructor_arguments).unwrap();
 }
 ```
+
+### Mocking the constructor with `roll`
```

Comment:
This is a little excessive, since if we wanted to be consistent with it, we'd have to document all of the cheatcodes that work in constructors in this way.

I would delete everything after this line for brevity.